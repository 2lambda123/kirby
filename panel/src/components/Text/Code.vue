<template>
	<!-- @slot Code content -->
	<pre class="k-code"><code><slot /></code></pre>
</template>

<script>
/**
 * @public
 */
export default {}
</script>

<style>
:root {
	--code-color-back: var(--color-black);
	--code-color-icon: var(--color-gray-500);
	--code-color-text: var(--color-gray-200, white);
	--code-font-family: var(--font-mono);
	--code-font-size: 1em;
}
.k-code,
.k-text pre {
	position: relative;
	display: block;
	font-family: var(--code-font-family);
	font-size: var(--code-font-size);
	max-width: 100%;
	padding: 0.75rem;
	border-radius: var(--rounded, 0.5rem);
	background: var(--code-color-back);
	color: var(--code-color-text);
	white-space: nowrap;
	overflow-y: hidden;
	overflow-x: auto;
	line-height: 1.5;
	tab-size: 2;
}
.k-code:not(code),
.k-text pre {
	white-space: pre-wrap;
}
.k-code::before {
	position: absolute;
	content: attr(data-language);
	inset-block-start: 0;
	inset-inline-end: 0;
	padding: 0.5rem 0.5rem 0.25rem 0.25rem;
	font-size: calc(0.75 * var(--text-xs));
	background: var(--code-color-back);
	border-radius: var(--rounded, 0.5rem);
}
</style>
