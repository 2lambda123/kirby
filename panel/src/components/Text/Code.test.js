import { describe, expect, it } from "vitest";
import { mount } from "@vue/test-utils";
import Code from "./Code.vue";

describe("Code.vue", () => {
	it("has default slot", async () => {
		const content = 'console.log("Hello, world!");';
		const wrapper = mount(Code, {
			slots: {
				default: content
			}
		});

		expect(wrapper.text()).toContain(content);
	});

	it("has CSS selector", async () => {
		const wrapper = mount(Code);
		expect(wrapper.classes()).toContain("k-code");
	});

	it("matches snapshot", async () => {
		const wrapper = mount(Code, {
			slots: {
				default: "This is some code"
			}
		});

		expect(wrapper.element).toMatchSnapshot();
	});
});
