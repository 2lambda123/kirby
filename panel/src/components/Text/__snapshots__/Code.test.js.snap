// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Code.vue > matches snapshot 1`] = `
<pre
  class="k-code"
>
  <code>
    This is some code
  </code>
</pre>
`;
